src/main/java/com/example/airportproject/service/FlightFetchService.java

```diff
@@ -0,0 +1,150 @@
+            // extract the array of flights from the JSON
+            JsonNode objectsArray = jsonNode.get("response");
+            // iterate over each flight in the array
+            for(JsonNode objectNode : objectsArray){
+                // get the current flight list (departures or arrivals)
+                List<Flight> fList = flightData.get(res);
+                // extract the flight details needed
+                // create a flight object for each flight in the response
+                // if list not empty - check that the previous flight is not duplicate
+                if(!fList.isEmpty()){
+                    Flight prev = fList.get(fList.size() - 1);
+                    // check prev flight does not have same dep airport, dep gate & dep time - duplicate of flight
+                    if(((prev.getDepIata().equals(objectNode.get("dep_iata").asText())) &&
+                            (prev.getDepGate().equals(objectNode.get("dep_gate").asText())) &&
+                            (prev.getDepTime().equals(OffsetDateTime.parse(objectNode.get("dep_time").asText().replace(" ", "T") + ":00+01")))
+                    )){
+                        continue;
+                    }
+                }
+                fList.add(new Flight(
+                        getTextValue(objectNode, "airline_iata"),
+                        getTextValue(objectNode, "dep_iata"),
+                        getTextValue(objectNode, "dep_terminal"),
+                        getTextValue(objectNode, "dep_gate"),
+                        getTextValue(objectNode, "arr_iata"),
+                        getTextValue(objectNode, "arr_terminal"),
+                        getTextValue(objectNode, "arr_gate"),
+                        getTextValue(objectNode, "status"),
+                        getTextValue(objectNode, "aircraft_icao"),
+                        getTextValue(objectNode, "flight_number"),
+                        getTextValue(objectNode, "flight_iata"),
+                        OffsetDateTime.parse(objectNode.get("dep_time").asText().replace(" ", "T") + ":00+01"),
+                        OffsetDateTime.parse(objectNode.get("arr_time").asText().replace(" ", "T") + ":00+01"),
+                        objectNode.get("duration").asInt()
+                ));
+            }
+        }
+        return flightData;
+    }
+
+    /**
+    * Fetches flight data from the external 'AirLabs' API, models the response to Flight objects, and persists the data to a Postgres database.
+     * Is called by the application runner (runs once when the application is started)
+     * @throws IOException
+     * @throws InterruptedException
+    */
+    public void fetchAndPersistFlights() throws IOException, InterruptedException {
+        String apiKey = "c5de155c-c17e-47fa-9eb1-500a6d74ffae"; // need to be stored securely
+        String airportCode = "MAN";
+        String apiUrl = "https://airlabs.co/api/v9/schedules?api_key=" + apiKey; // provide api key
```

Comment:
It might be better to a) store these variables in your application properties for easy access, and b) not push secrets like your api key to GitHub as other will now have access to it.